drgn/helpers/common/memory.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Memory
+------
+
+The ``drgn.helpers.memory`` module provides helpers for working with memory and addresses.
+"""
+
+import operator
+from typing import Optional, Union, overload
+
+import drgn
+from drgn import IntegerLike, Object, Program, SymbolKind
+from drgn.helpers.common.format import escape_ascii_string
+from drgn.helpers.linux.slab import find_containing_slab_cache
+
+__all__ = ("identify_address",)
+
+
+_SYMBOL_KIND_STR = {
+    SymbolKind.OBJECT: "object symbol",
+    SymbolKind.FUNC: "function symbol",
+}
+
+
+@overload
+def identify_address(addr: Object) -> Optional[str]:
+    """"""
+    ...
+
+
+@overload
+def identify_address(prog: Program, addr: IntegerLike) -> Optional[str]:
+    """
+    Identify an address (given as a :class:`.Program` and an integer) as a symbol or slab object.
+
+    This will currently identify the following types of addresses as follows:
+
+    * Symbols (e.g., addresses in functions or global variables): `symbol_kind: symbol_name+offset`.
+    * Slab objects: `slab object: slab_cache_name`.
+
+    This may learn to recognize other types of addresses in the future.
```

Comment:
Looking at this again, it could use more clarification:

```suggestion
    For all programs, this will identify addresses as follows:

    * Object symbols (e.g., addresses in global variables):
      ``object symbol: {symbol_name}+{hex_offset}``.
    * Function symbols (i.e., addresses in functions):
      ``function symbol: {symbol_name}+{hex_offset}``.
    * Other symbols: ``symbol: {symbol_name}+{hex_offset}``.

    Additional, for the Linux kernel, this will identify:

    * Slab objects: ``slab object: {slab_cache_name}``.

    This may recognize other types of addresses in the future.
```
```

Reply:
Expanded!